### ISSUE TITLE
Request-Transformer: Unable to add values that contain colons

### ISSUE THREAD

**Issue by ThorArakis** [Mashape/kong#1287]
I would like to add arbitrary strings that include colons in them into the body (and also potentially the head). When I try to add something like `"scopes": "scope1:scope2"` through form data the resulting plugin configuration looks like:
```
"config": {
    "add": {
      "body": [
        "scopes:scope1:scope2"
      ],
      "querystring": {},
      "headers": [
        "foo:bar:baz",
      ]
    },
  }
```
But the output is only `scope1` gets added to the body:
```
"json": {
    "scopes": "scope1"
  }
```

This is similar behavior between the headers and the body transforms.

**Comment by thefosk** [comment]
The problem is here: https://github.com/Mashape/kong/blob/master/kong/plugins/request-transformer/access.lua#L66

`current_value` should be the combination of all the other elements, not just the first one. The same problem also happens in the Response Transformer: https://github.com/Mashape/kong/blob/master/kong/plugins/response-transformer/body_transformer.lua#L41

**Comment by Tieske** [comment]
fixed it in #1289

**Comment by DonMartin76** [comment]
Ha, just wanted to enter a new issue for this, as it just bit me. I wanted to add a `X-ApiHost` header with the "outside" URL for an API (btw, why don't I get the `X-Forwarded-Host` back?), but it fails at both the colon on `https://` and the port (e.g. `https://api.mycompany.com:8443/whatever`). Will this make it into one of the nearer releases?

**Comment by Tieske** [comment]
Sorry for the delays. We're in a big refactor, which is taking a lot of resources. Not sure it will make the next release, but certainly the one after.

**Comment by jmdacruz** [comment]
is there any workaround for this in the meantime?

**Comment by Tieske** [comment]
fixed in #1353, will be in 0.9
